q2as_json.h

```diff
@@ -9,6 +9,139 @@
 
 #include <string_view>
 
+template<typename TargetType, typename SourceType>
+bool q2as_type_in_range(SourceType value)
+{
+    // Prevent bool as TargetType or SourceType to avoid edge cases
+    static_assert(!std::is_same_v<TargetType, bool>, "TargetType cannot be bool");
```

Comment:
prob quick-return true in case TargetType and SourceType are the same

Reply:
Fixed.